pkg/client/fake/doc.go

```diff
@@ -17,17 +17,24 @@ limitations under the License.
 /*
 Package fake provides a fake client for testing.
 
-Deprecated: please use pkg/envtest for testing. This package will be dropped
-before the v1.0.0 release.
-
-An fake client is backed by its simple object store indexed by GroupVersionResource.
+fake client is backed by its simple object store indexed by GroupVersionResource.
 You can create a fake client with optional objects.
 
 	client := NewFakeClient(initObjs...) // initObjs is a slice of runtime.Object
 
 You can invoke the methods defined in the Client interface.
 
-When it doubt, it's almost always better not to use this package and instead use
+When in doubt, it's almost always better not to use this package and instead use
 envtest.Environment with a real client and API server.
+
+Current Limitations / Known Issues with the fake Client:
+- this client does not use reactors so it can not be setup with detailed responses for testing
+- possible locking issues when using fake client and not the actual client
```

Comment:
Which locking issues are we referring to? It'd be great to have more details

Reply:
I noted this down as we were discussing in the sig meeting so you are right we need more detail on this item. Are there others that can provide more detail here?

Reply:
I'd double check with the recording, but IIRC this is related to locking with update/patch when using resourceVersion, which is covered below

Reply:
You are right. Changed it and resolving this.